src/pages/today/index.tsx

```diff
@@ -25,32 +58,63 @@ export type HabitTodo = {
 export default function Today() {
 	const [date, setDate] = useState(new Date());
 	const [habitTodos, setHabitTodos] = useState<HabitTodo[]>([]);
+	const documentId = useRef<string>(null);
 
 	const completedHabits = habitTodos.filter((habit) => habit.isComplete);
 	const incompleteHabits = habitTodos.filter((habit) => !habit.isComplete);
 
 	const isDateToday = isToday(date);
 	const isDateYesterday = isYesterday(date);
 
+	const handleCheckHabit =
+		(habitId: string) => (e: React.ChangeEvent<HTMLInputElement>) =>
+			markHabitComplete(e.target.checked, habitId, documentId.current);
+
 	useEffect(() => {
-		const unsub = onSnapshot(
-			getDateDoc(date),
-			(doc) => {
-				if (doc.exists()) {
-					const { date, ...habits } = doc.data();
-					setHabitTodos(
-						Object.entries(habits).map(([id, habit]) => ({
-							id,
-							...(habit as Omit<HabitTodo, 'id'>),
-						}))
+		return onSnapshot(
```

Comment:
This useEffect is pretty long and hard to follow. I'd definitely recommend refactoring it, or at the very least just write a summary comment of what is happening in here. 

A couple rules I try to follow: 
* I usually have a lint rule that warns me for functions that are longer than 12 lines. So this would be a red flag
* Functions should be easily understood to someone reading the code. If it takes more than a minute to parse through and understand what a function is doing that is a red flag. 
* If I can't refactor it for some reason, add a summary block comment that explains what is happening in plain text

Reply:
A good place to start is looking at the if/else (lines 82 - 115)

maybe something like this:

```
      if (empty) {
        await createNewDateDocument(date);
      } else {
        await handleExistingDateDocument(docs[0], setHabitTodos);
      }
```

Reply:
I created a separate function for creating a date document to improve readability.